src/libcore/cell.rs

```diff
@@ -1548,6 +1548,36 @@ impl<T: ?Sized> UnsafeCell<T> {
         // #[repr(transparent)]
         self as *const UnsafeCell<T> as *const T as *mut T
     }
+
+    /// Gets a mutable pointer to the wrapped value.
+    ///
+    /// This can be cast to a pointer of any kind.
+    /// Ensure that the access is unique (no active references, mutable or not)
+    /// when casting to `&mut T`, and ensure that there are no mutations
+    /// or mutable aliases going on when casting to `&T`
+    ///
+    /// # Examples
+    ///
+    /// Gradual initialization of an `UnsafeCell`:
+    ///
+    /// ```
+    /// #![feature(unsafe_cell_raw_get)]
+    /// use std::cell::UnsafeCell;
+    /// use std::mem::MaybeUninit;
+    ///
+    /// let m = MaybeUninit::<UnsafeCell<i32>>::uninit();
+    /// unsafe { m.as_ptr().raw_get().write(5); }
+    /// let uc = unsafe { m.assume_init() };
+    ///
+    /// assert_eq!(uc.into_inner(), 5);
+    /// ```
+    #[inline]
+    #[unstable(feature = "unsafe_cell_raw_get", issue = "0")]
+    pub const fn raw_get(self: *const Self) -> *mut T {
```

Comment:
This should be an associated function since `self: *const Self` isn't stable yet.

Reply:
Could it be an inherent method of `Self = *const UnsafeCell<T>` ?

Reply:
We also had a stable `Future` trait before users could use `Pin` as a `self` type, didn't we?

Reply:
No, `Pin` as receiver came first (1.33.0) and then came `Future` (1.36.0) -- I'm personally not enthusiastic about `self: *const Self` (which doesn't have an accepted RFC yet also) and at any rate, it would be good for the documentation to reflect actual expected usage patterns.

Reply:
Is it fuutre-compatible to make this a method later? This definitely wants to be a method to get postfix syntax. Raw pointers are already unergonomic enough, denying them method syntax is one speed-bump too much.

Reply:
> No, Pin as receiver came first (1.33.0) and then came Future (1.36.0) 

Also note that this is an unstable feature that we are adding here -- `Future` certainly landed unstable before `Pin`-receiver was stable.

Reply:
> Also note that this is an unstable feature that we are adding here -- `Future` certainly landed unstable before `Pin`-receiver was stable.

Yeah but why set roadblocks for eventual stabilization. ^^

I believe it should be forward compat given that a method is callable with UFCS. @danielhenrymantilla's idea won't work because: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=fdd74e8936e23afd8f74b238bbdb309b

Reply:
I made it an assoc function and added a note in the tracking issue.